perfkitbenchmarker/linux_packages/maven.py

```diff
@@ -70,22 +96,48 @@ def _Install(vm):
   maven_full_ver = FLAGS.maven_version
   maven_major_ver = maven_full_ver[:maven_full_ver.index('.')]
   maven_url = MVN_URL.format(maven_major_ver, maven_full_ver)
-  vm.RemoteCommand(
-      ('mkdir {0} && curl -L {1} | '
-       'tar -C {0} --strip-components=1 -xzf -').format(MVN_DIR, maven_url))
-
-  # Get JAVA_HOME
-  out, _ = vm.RemoteCommand('readlink -f `which java`')
-  out = out.strip()
-  java_home = out[:out.index('/jre')]
+  maven_tar = maven_url.split('/')[-1]
+  if maven_tar not in PREPROVISIONED_DATA:
+    PREPROVISIONED_DATA[maven_tar] = ''  # will only work with preprovision_ignore_checksum
+    PACKAGE_DATA_URL[maven_tar] = maven_url
+  maven_remote_path = posixpath.join(INSTALL_DIR, maven_tar)
+  vm.InstallPreprovisionedPackageData(
+      PACKAGE_NAME,
+      [maven_tar],
+      INSTALL_DIR
+  )
+  vm.RemoteCommand(('mkdir -p {0} && '
+                    'tar -C {0} --strip-components=1 -xzf {1}').format(
+                        MVN_DIR, maven_remote_path))
+
+  java_home = _GetJavaHome(vm)
 
   # Set env variables for maven
   maven_env = MVN_ENV.format(java_home=java_home, maven_home=MVN_DIR)
   cmd = 'echo "{0}" | sudo tee -a {1}'.format(maven_env, MVN_ENV_PATH)
   vm.RemoteCommand(cmd)
 
+  # On CentOS, the mvn command looks for tools.jar in $JAVA_HOME/../lib
+  java_lib = posixpath.join(java_home, 'lib')
+  java_lib_sym = posixpath.join(java_home, '..', 'lib')
+  vm.RemoteCommand('sudo ln -sf {0} {1}'.format(java_lib, java_lib_sym))
+
+  if FLAGS.maven_mirror_url != '':
+    settings_local_path = data.ResourcePath(posixpath.join(
+        'maven', 'settings.xml.j2'))
+    settings_remote_path = '~/.m2/settings.xml'
+    context = {
+        'maven_mirror_url': FLAGS.maven_mirror_url
+    }
+    vm.RemoteCommand('mkdir -p ~/.m2')
+    vm.RenderTemplate(settings_local_path, settings_remote_path, context)
+
 
 def Uninstall(vm):
+  java_home = _GetJavaHome(vm)
+  java_lib_sym = posixpath.join(java_home, '..', 'lib')
+  vm.RemoteCommand('sudo rm -f {0}'.format(java_lib_sym))
   vm.Uninstall('openjdk')
+  vm.Uninstall('curl')
```

Comment:
What if the OS came with curl? seems better to leave it.

Reply:
@pmkc fair enough. I have removed this line.